### PR TITLE
Check if index page exists when updating slug

### PR DESCRIPTION
**Pull request by denmit**

### CODE REVIEW

File: Handler/SlugHandler.php [hunk 1]
```diff
@@ -52,4 +55,48 @@ private function appendIndex()
...
+        
+        $this->_checkSlugIndex();
+    }
+    
+    /**
+     * Loop through all pages with same slug and increment by one
+     * 
+     * @param int $i
+     */
+    private function _checkSlugIndex($i = 0)
```
Comment (rvanlaarhoven) [inline Handler/SlugHandler.php]: Why does this method start with a `_`?

**Comment by rvanlaarhoven** [comment]
Can you explain what this PR actually does? Does it try to avoid duplicate `/../index` slugs?

**Comment by denmit** [comment]
Whenever a new slug is generated, SlugHandler checks to see if  `/slug/index` already exists, and if it does it gets appended with incrementing number such as: `slug-1` and goes through whole process again.

This is for ContentRouter to always redirect to index pages if `/slug` does not exist.

File: Handler/SlugHandler.php (hunk 1 above)
Comment (denmit) [inline Handler/SlugHandler.php]: For private methods, but doesn't make sense here, so I'll remove it.

**Comment by rvanlaarhoven** [comment]
Isn't this already done with that Gedmo SlugHandler? How does it work for other non-index slugs?

If not, I think it's more efficient to simply check how many matches there are, then get the one with the highest number at the end (if any) and just put that number ++.

See for example how it's done here: https://github.com/Opifer/MediaBundle/blob/master/Provider/FileProvider.php#L191-L205

Thinking about this; could you also call the method `createUniqueIndexSlug` or something?

**Comment by denmit** [comment]
The idea is to have ContentRouter always serve index pages, if any, for `/slug` and `/slug/` urls.

Since index slugs get saved as `directory-slug/index` any new content slugs that are same as directory-slug could potentially serve different content for  `/slug` and `/slug/` urls.

So the method checks for `/slug/index`, `/slug-1/index` ... `/slug-n/index`, then saves the slug as `/slug-n`

Maybe there is something similar in Gedmo SlugHandler already, i'll check.

**Comment by dylanschoenmakers** [comment]
The ContentRouter should not serve ``/slug`` for ``/slug/index``, but redirect 302 to ``/slug/`` instead if there's no hit on ``/slug`` itself.

**Comment by denmit** [comment]
ContentRouter does just that, so this method tries to avoid saving of `/slug` if `/slug/index` already exists.

This is useful for nested content where slug can not be edited easily, but a better solution might be to append content id to nested content slug.
I'll refactor this pull request to support this.

**Comment by denmit** [comment]
Closing this pull request.

**Pull request closed without merging**